jac/jaclang/compiler/passes/main/import_pass.py

```diff
@@ -169,28 +170,68 @@ def import_jac_mod_from_dir(self, target: str) -> ast.Module | None:
                 kid=[ast.EmptyToken()],
             )
 
+    def load_cached_module(
+        self, source_path: str, cache_path: str
+    ) -> ast.Module | None:
+        """Load cached module if up-to-date."""
+        if not os.path.exists(cache_path):
+            return None
+
+        try:
+            source_mtime = os.path.getmtime(source_path)
+            cache_mtime = os.path.getmtime(cache_path)
```

Comment:
we don't check the `mtime` of the cached file instead of this store the mtime of the source file within the pickle dump and check with the current `mtime` of the source file